middleware_retry.go

```diff
@@ -26,11 +24,7 @@ func retryProcessError(queue string, mgr *Manager, message *Msg, err error) erro
 			) * time.Second,
 		)
 
-		rc := mgr.opts.client
-		_, err = rc.ZAdd(mgr.RetryQueue(), redis.Z{
-			Score:  nowToSecondsWithNanoPrecision() + waitDuration,
-			Member: message.ToJson(),
-		}).Result()
+		err := mgr.opts.store.EnqueueRetriedMessage(nowToSecondsWithNanoPrecision()+waitDuration, message.ToJson())
```

Comment:
we may want `err =` here instead of `err :=`, otherwise we subtly change the behavior of the return here

Reply:
Good catch, will fix it.